hosts/darwin02/builder.nix

```diff
@@ -0,0 +1,25 @@
+{
+  users.knownGroups = [ "nix" ];
+  users.knownUsers = [ "nix" ];
+
+  users.groups.nix = {
+    name = "nix";
+    gid = 8765;
+    description = "Group for remote build clients";
+  };
+
+  users.users.nix = {
+    name = "nix";
+    uid = 8765;
+    home = "/Users/nix";
+    createHome = true;
+    shell = "/bin/zsh";
+    description = "User for remote build clients";
+    # keys are copied, not symlinked
+    openssh.authorizedKeys.keys = [
+      "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIEmdo1x1QkRepZf7nSe+OdEWX+wOjkBLF70vX9F+xf68 builder"
```

Comment:
It's possible to further restrict the access so the ssh user can only be used for remote building. See: 
https://github.com/numtide/srvos/blob/4f22e6fcaf17c6313c2ecdc996760c3e4b14a623/nixos/roles/nix-remote-builder.nix#LL32C8-L32C8

Reply:
If we add darwin support to `srvos` I'm planning to adjust that module support darwin and then using it here for both darwin and linux.